tool/generator/doc_provider.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
```

Comment:
optional: It may be useful to combine this file with bcd.dart so that it can share some common patterns if any.